### ISSUE TITLE
Make RobotRules accessible

### ISSUE THREAD

**Issue by naegelejd** [crawler-commons/crawler-commons#134]
It would be very handy to have access to the set of internal `RobotRule`s in `SimpleRobotRules`.

Either a "getter", or just changing its access modifier from `private` to `protected` ([here](https://github.com/crawler-commons/crawler-commons/blob/eba560af1c787fe7f5d88f36645c921a28639f5c/src/main/java/crawlercommons/robots/SimpleRobotRules.java#L106)) so I can implement it myself. Something like that.

I'd like to be able to extract the rules and do what I want with them. In other words, it would be good to better separate *parsing* and *matching* rules.

**Comment by kkrugler** [comment]
Feels like we'd want a BaseRobotRule then, though it sounds like you'd rather hard-code your stuff to the concrete implementation of SimpleRobotRules.

**Comment by naegelejd** [comment]
Yeah I understand this is kind of an architectural change. I think it makes sense though: a language parser, for example, might produce an abstract syntax tree, which can then be used to do many different things.

But yes, you're correct, I just want the SimpleRobotRules. For my use case, the parser and resulting rules are very useful, and the `isAllowed(url)` functionality is too restrictive.

**Comment by jnioche** [comment]
I like the idea of separating the extraction of the rules from the selection of which ones should be applied.

**Comment by aecio** [comment]
Besides not being accessible, `RobotRules` doesn't have a no-argument public constructor, which creates problems for serialization libraries. This prevents crawlers from caching robot rules. Would you accept a pull request adding these changes?

**Comment by kkrugler** [comment]
@aecio - yes, happy to accept a PR for this.
